packages/core/src/__tests__/button.spec.ts

```diff
@@ -62,9 +62,12 @@ describe('Button', () => {
     expect(onClick).toHaveBeenCalledOnce();
   });
 
-  it('Renders a disabled button', () => {
-    render(Button, { disabled: true });
-    expect(screen.getByRole('button')).toBeDisabled();
+  it('Renders a disabled button', async () => {
+    const { component } = render(Button, { disabled: true });
+    const onClick = vi.fn();
+    component.$on('click', onClick);
+    await fireEvent.click(screen.getByRole('button'));
+    expect(onClick).not.toHaveBeenCalledOnce();
```

Comment:
[nit] if `onClick` was called twice, this assertion would pass

```suggestion
    expect(onClick).not.toHaveBeenCalled();
```

Reply:
Whoops that is an autocomplete miss, I meant to just use `toHaveBeenCalled` good catch.